File: ar601_trajectories/src/f_z.c
#include "rt_nonfinite.h"
#include "main_trajectory.h"
#include "f_z.h"
#include "main_data.h"

double f_z(double t, double up)
{
  double z;
  if (up == 1.0) {
    if (t <= p_ds * t0) {
      z = 0.0;
    } else {
      z = 0.5 * h * (1.0 - cos(2.0 * (1.0 / (t0 - p_ds * t0)) *
        3.1415926535897931 * (t - p_ds * t0)));
    }
  } else {
    z = 0.0;
  }

  return z;
}

File: ar601_trajectories/generation_octave/f_z.m
function [ z ] = f_z(t,up)

global t0 p_ds h

if up==1
    
    if t<=p_ds*t0
        z=0;
    else
        z=0.5*h*(1-cos(2*(1/(t0-p_ds*t0))*pi*(t-p_ds*t0)));
    end
    
else
    z=0;
end

end

